substrafl/algorithms/algo.py

```diff
@@ -1,10 +1,15 @@
+from __future__ import annotations
```

Comment:
do we need both this and the TYPE_CHECKING trick?

Reply:
Yes, it avoid changing the type hint "strategy name" to string:
`This isn’t necessary if we add from __future__ import annotations at the top of the file, which makes all annotations strings by default (see [PEP 563](https://www.python.org/dev/peps/pep-0563/#abstract)). (__future__.annotations will become the default in Python 3.11).`